staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -98,7 +99,7 @@ func (d *YAMLToJSONDecoder) Decode(into interface{}) error {
 	}
 
 	if len(bytes) != 0 {
-		err := yaml.Unmarshal(bytes, into)
+		err := yaml3.Unmarshal(bytes, into)
```

Comment:
Patch this piece of logic separately and introduce new dependencies. I don't think this is a good modification.